src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1232,11 +1238,61 @@ public void ClearCurrentWorkspace()
 
             OnWorkspaceCleared(this, EventArgs.Empty);
         }
+
+        public IEnumerable<LibraryItem> GetAllLibraryItemsByCategory()
+        {
+            if (allLibraryItems == null || allLibraryItems.Count == 0)
+            {
+                allLibraryItems = new List<LibraryItem>();
+                foreach (var elem in this.SearchModel.SearchEntries)
+                {
+                    var libItem = new LibraryItem(elem);
+                    libItem.Keywords = SearchModel.GetTags(elem);
+                    SetPorts(libItem);
+                    allLibraryItems.Add(libItem);
+                }
+            }
+
+            return allLibraryItems;
+        }
         
         #endregion
 
         #region private methods
 
+        private void SetPorts(LibraryItem item)
+        {
+            var functionItem = LibraryServices.GetFunctionDescriptor(item.CreationName);
+            NodeModel newElement = null;
+            if (functionItem != null)
+            {
+                item.DisplayName = functionItem.DisplayName;
+                newElement = (functionItem.IsVarArg)
+                    ? new DSVarArgFunction(functionItem) as NodeModel
+                    : new DSFunction(functionItem);
+            }
+            // If that didn't work, let's try using the NodeFactory
+            else
+            {
+                NodeFactory.CreateNodeFromTypeName(item.CreationName, out newElement);
+            }
+
+            if (newElement == null)
+                throw new TypeLoadException("Unable to create instance of NodeModel element by CreationName");
+
+            item.Parameters = newElement.InPorts.Select(elem => new LibraryItem.PortInfo
+            {
+                Name = elem.PortName,
+                Type = elem.ToolTipContent.Split('.').Last(),
+                DefaultValue = newElement.InPortData[elem.Index].DefaultValue
+            });
+
+            item.ReturnKeys = newElement.OutPorts.Select(elem => new LibraryItem.PortInfo
+            {
+                Name = elem.PortName
+            });
+        }
+
```

Comment:
@pboyer, I remember asking this to be moved out of `LibraryServices` (or something similar) because we started passing `DynamoModel` everywhere just to do simple things such as these. If it doesn't belong here it definitely belongs to some library related classes owned by `DynamoModel`.